physical/postgresql.go

```diff
@@ -0,0 +1,194 @@
+package physical
+
+import (
+	"database/sql"
+	"fmt"
+	"sort"
+	"strings"
+	"time"
+
+	"github.com/armon/go-metrics"
+	_ "github.com/lib/pq"
+)
+
+// PostgreSQL Backend is a physical backend that stores data
+// within a PostgreSQL database.
+type PostgreSQLBackend struct {
+	table      string
+	client     *sql.DB
+	statements map[string]*sql.Stmt
+}
+
+// newPostgreSQLBackend constructs a PostgreSQL backend using the given
+// API client, server address, credentials, and database.
+func newPostgreSQLBackend(conf map[string]string) (Backend, error) {
+	// Get the PostgreSQL credentials to perform read/write operations.
+	connURL, ok := conf["connection_url"]
+	if !ok || connURL == "" {
+		return nil, fmt.Errorf("missing connection_url")
+	}
+
+	table, ok := conf["table"]
+	if !ok {
+		table = "vault"
+	}
+
+	// Create PostgreSQL handle for the database.
+	db, err := sql.Open("postgres", connURL)
+	if err != nil {
+		return nil, fmt.Errorf("failed to connect to postgres: %v", err)
+	}
+
+	// Create the required table if it doesn't exists.
+	create_query := "CREATE TABLE IF NOT EXISTS " + table +
+		" (vault_key VARCHAR(512), vault_value BYTEA, PRIMARY KEY (vault_key))"
```

Comment:
Is there an advantage to using `VARCHAR` here? From what I understand `TEXT` is just as efficient as `VARCHAR` in postgres:

> Tip: There is no performance difference among these three types, apart from increased storage space when using the blank-padded type, and a few extra CPU cycles to check the length when storing into a length-constrained column. While character(n) has performance advantages in some other database systems, there is no such advantage in PostgreSQL; in fact character(n) is usually the slowest of the three because of its additional storage costs. In most situations text or character varying should be used instead.

http://www.postgresql.org/docs/9.1/static/datatype-character.html